pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -114,42 +116,33 @@ func (c ConfigObserver) sync() error {
 	if !equality.Semantic.DeepEqual(existingConfig, mergedObservedConfig) {
 		glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(existingConfig, mergedObservedConfig))
 		spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: mergedObservedConfig}}
-		_, resourceVersion, err = c.operatorClient.UpdateOperatorSpec(resourceVersion, spec)
+		_, resourceVersion, err = c.operatorConfigClient.UpdateOperatorSpec(resourceVersion, spec)
 		if err != nil {
 			errs = append(errs, fmt.Errorf("error writing updated observed config: %v", err))
 			c.eventRecorder.Warningf("ObservedConfigWriteError", "Failed to write observed config: %v", err)
 		} else {
 			c.eventRecorder.Eventf("ObservedConfigChanged", "Writing updated observed config")
 		}
 	}
+	err = common.NewMultiLineAggregate(errs)
 
-	status := originalStatus.DeepCopy()
-	if len(errs) > 0 {
-		var messages []string
-		for _, currentError := range errs {
-			messages = append(messages, currentError.Error())
-		}
-		v1helpers.SetOperatorCondition(&status.Conditions, operatorv1.OperatorCondition{
-			Type:    operatorStatusTypeConfigObservationFailing,
-			Status:  operatorv1.ConditionTrue,
-			Reason:  configObservationErrorConditionReason,
-			Message: strings.Join(messages, "\n"),
-		})
-
-	} else {
-		v1helpers.SetOperatorCondition(&status.Conditions, operatorv1.OperatorCondition{
-			Type:   operatorStatusTypeConfigObservationFailing,
-			Status: operatorv1.ConditionFalse,
-		})
+	// update failing condition
+	cond := operatorv1.OperatorCondition{
+		Type:   operatorStatusTypeConfigObservationFailing,
+		Status: operatorv1.ConditionFalse,
 	}
-
-	if !equality.Semantic.DeepEqual(originalStatus, status) {
-		_, _, err = c.operatorClient.UpdateOperatorStatus(resourceVersion, status)
-		if err != nil {
-			return err
-		}
+	if err != nil {
+		cond.Status = operatorv1.ConditionTrue
+		cond.Reason = "Error"
+		cond.Message = err.Error()
+	}
+	v1helpers.SetOperatorCondition(&operatorStatus.Conditions, cond)
+	if _, updateError := c.updateStatus(resourceVersion, originalOperatorStatus, operatorStatus); updateError != nil {
```

Comment:
why doesn't this one use your new retrying updateSTatus method?

Reply:
done